src/suggestionlistworker.cpp

```diff
@@ -48,13 +30,20 @@ void SuggestionListWorker::run()
         QUrl url;
         url.setScheme("zim");
         url.setHost(currentZimId + ".zim");
-        kiwix::SuggestionsList_t suggestions;
-        searchSuggestionsSmart(archive, m_text.toStdString(), 15, suggestions);
-        for (auto& suggestion: suggestions) {
-            QString path = QString("/") + QString::fromStdString(suggestion.getPath());
-            url.setPath(path);
-            suggestionList.append(QString::fromStdString(suggestion.getTitle()));
-            urlList.append(url);
+        int suggestionsCount = 15;
+        auto prefix = m_text.toStdString();
+        auto suggestionSearcher = zim::SuggestionSearcher(*archive);
+        if (archive->hasTitleIndex()) {
+            auto suggestionSearch = suggestionSearcher.suggest(prefix);
+            const auto suggestions = suggestionSearch.getResults(0, suggestionsCount);
+            for (auto current : suggestions) {
+                kiwix::SuggestionItem suggestion(current.getTitle(), lcAll(current.getTitle()),
```

Comment:
Do we really need to use a `kiwix::SuggestionItem` ?
Could we feed suggestionList, url and urlList directly from `current` ?